File: test_data/csharp/test_main/aas_core_meta.v3rc2/input/snippets/Verification/data_specification_IEC_61360s_for_reference_have_appropriate_data_type.cs
/// <summary>
/// Check that the <see cref="Aas.DataSpecificationIec61360.DataType" /> is defined
/// appropriately for all data specifications whose content is given as IEC 61360.
/// </summary>
[CodeAnalysis.SuppressMessage("ReSharper", "InconsistentNaming")]
public static bool DataSpecificationIec61360sForReferenceHaveAppropriateDataType(
    IEnumerable<Aas.EmbeddedDataSpecification> embeddedDataSpecifications
)
{
    foreach (var embeddedDataSpecification in embeddedDataSpecifications)
    {
        var iec61360 = (
            embeddedDataSpecification.DataSpecificationContent
                as DataSpecificationIec61360
        );
        if (iec61360 != null)
        {
            if (
                iec61360.DataType == null
                || !Constants.DataTypeIec61360ForReference.Contains(
                    iec61360.DataType)
            )
            {
                return false;
            }
        }
    }

    return true;
}

File: test_data/typescript/test_main/aas_core_meta.v3rc2/input/snippets/Verification/data_specification_IEC_61360s_for_reference_have_appropriate_data_type.ts
/**
 * Check that {@link types.DataSpecificationIec61360.dataType}
 * is defined appropriately for all data specifications whose content is given as
 * IEC 61360.
 *
 * @param embeddedDataSpecifications - to be verified
 * @returns `true` if the check passes
 */
export function dataSpecificationIec61360sForReferenceHaveAppropriateDataType(
  embeddedDataSpecifications: Iterable<AasTypes.EmbeddedDataSpecification>
): boolean {
  for (const embeddedDataSpecification of embeddedDataSpecifications) {
    const content = embeddedDataSpecification.dataSpecificationContent;
    if (AasTypes.isDataSpecificationIec61360(content)) {
      if (
        content.dataType === null
        || !AasConstants.DATA_TYPE_IEC_61360_FOR_REFERENCE.has(content.dataType)
      ) {
        return false;
      }
    }
  }

  return true;
}
